test/unit/BCoWHelper.t.sol

```diff
@@ -0,0 +1,180 @@
+    // it should return pool tokens
+    address[] memory returned = helper.tokens(address(pool));
+    assertEq(returned[0], tokens[0]);
+    assertEq(returned[1], tokens[1]);
+  }
+
+  function test_OrderRevertWhen_ThePoolIsNotSupported() external {
+    // it should revert
+    vm.expectRevert(ICOWAMMPoolHelper.PoolDoesNotExist.selector);
+    helper.order(invalidPool, priceVector);
+  }
+
+  function test_OrderWhenThePoolIsSupported(bytes32 domainSeparator) external {
+    // it should query the domain separator from the pool
+    pool.expectCall_SOLUTION_SETTLER_DOMAIN_SEPARATOR();
+    pool.mock_call_SOLUTION_SETTLER_DOMAIN_SEPARATOR(domainSeparator);
+
+    (
+      GPv2Order.Data memory order_,
+      GPv2Interaction.Data[] memory preInteractions,
+      GPv2Interaction.Data[] memory postInteractions,
+      bytes memory sig
+    ) = helper.order(address(pool), priceVector);
+
+    // it should return a valid pool order
+    assertEq(order_.receiver, GPv2Order.RECEIVER_SAME_AS_OWNER);
+    assertLe(order_.validTo, block.timestamp + 5 minutes);
+    assertEq(order_.feeAmount, 0);
+    assertEq(order_.appData, factory.APP_DATA());
+    assertEq(order_.kind, GPv2Order.KIND_SELL);
+    assertEq(order_.buyTokenBalance, GPv2Order.BALANCE_ERC20);
+    assertEq(order_.sellTokenBalance, GPv2Order.BALANCE_ERC20);
+
+    // it should return a commit pre-interaction
+    assertEq(preInteractions.length, 1);
+    assertEq(preInteractions[0].target, address(pool));
+    assertEq(preInteractions[0].value, 0);
+    bytes memory commitment = abi.encodeCall(IBCoWPool.commit, GPv2Order.hash(order_, domainSeparator));
+    assertEq(keccak256(preInteractions[0].callData), keccak256(commitment));
+
+    // it should return an empty post-interaction
+    assertTrue(postInteractions.length == 0);
+
+    // it should return a valid signature
+    bytes memory validSig = abi.encodePacked(pool, abi.encode(order_));
+    assertEq(keccak256(validSig), keccak256(sig));
+  }
+
+  function test_OrderGivenAPriceVector(uint256 priceSkewness, uint256 balanceToken0, uint256 balanceToken1) external {
+    // skew the price by max 50% (more could result in reverts bc of max swap ratio)
+    priceSkewness = bound(priceSkewness, 5000, 15_000);
+    vm.assume(priceSkewness != 10_000); // avoids no-skewness revert
+
+    balanceToken0 = bound(balanceToken0, 1e18, 1e36);
+    balanceToken1 = bound(balanceToken1, 1e18, 1e36);
+    vm.mockCall(tokens[0], abi.encodePacked(IERC20.balanceOf.selector), abi.encode(balanceToken0));
+    vm.mockCall(tokens[1], abi.encodePacked(IERC20.balanceOf.selector), abi.encode(balanceToken1));
+
+    uint256[] memory prices = new uint256[](2);
+    prices[0] = balanceToken1;
+    prices[1] = balanceToken0 * priceSkewness / 10_000;
+
+    // it should return a valid pool order
+    (GPv2Order.Data memory ammOrder,,,) = helper.order(address(pool), prices);
+
+    assertEq(address(ammOrder.buyToken), priceSkewness > 10_000 ? tokens[0] : tokens[1]);
```

Comment:
are we sure these are not different branches?

Reply:
wdym by "not different branches"? 
i think is more of a different approach but same branch, `test_OrderWhenThePoolIsSupported` doesn't check the amounts and just use the default price vector (`1:1.05`), while `test_OrderGivenAPriceVector` fuzzes both balances and the skeweness but checks only that the pool can verify it, and that the token being sold is the correct one.

we could entirely fuzz `test_OrderWhenThePoolIsSupported`, but we'd be adding complexity to that test instead of maintaining the "valid scenario" approach, dunno wdyt of that

Reply:
I believe that to fully benefit from btt tests we should make the `.tree` fully match the branches available in the (code|domain). I've checked and they're effectively different branches: `node_modules/cow-amm/src/libraries/GetTradeableOrder.sol:56`

It was not obvious for me reviewing it, and it could have been easier to see had the information of `when token0 reserve * denominator > when token1 reserve * numerator` -> `buyToken is token1` been in the `.tree` file. I believe this is a case where you set some 'funny values' to get to that branch. I consider this PR should include:

- [x] fully fuzzed tests to report on rounding errors (this one is ok for these purposes, we could also be purists and have one fully-fuzzed for each branch. Definetly not required.) This doesnt fit neatly into the .tree but I can live with that.
- [ ] 'valid scenario' 'funny value'd unit tests that are described in .tree and walk every available branch. In this case I'd consider the library an implementaiton detail and walk their branches as well, as that's critical to ensure this helper works as intended. This includes:
    - [ ] one testcase forcing token0 to be buyToken
    - [ ] one testcase forcing token1 to be buyToken
    - [ ] Some assertions on `src/contracts/BCoWHelper.sol:74`, since that defines two branches as well. Since they are coupled to the above, I don't know if they'll be their own separate two testcases or extra asserts inside them, do whatever feels right.

tl;dr: fuzzing is ok as is, we should have tests manually walking each branch.